src/Queue/Job.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Queue;
+
+use Symbiote\QueuedJobs\Services\AbstractQueuedJob;
+use Symbiote\QueuedJobs\Services\QueuedJob;
+
+/**
+ * Class Job
+ *
+ * This job represent a generic job which consumes items and each item is processed in one step
+ * useful as a template for jobs which need to process multiple items
+ * item can be an arbitrary piece of data for example: ID or array
+ *
+ * @property array $items
+ * @property array $remaining
+ * @package App\Queue
+ */
+abstract class Job extends AbstractQueuedJob
+{
+
+    public function getJobType(): int
+    {
+        return QueuedJob::QUEUED;
+    }
+
+    public function setup(): void
+    {
+        $this->remaining = $this->items;
+        $this->totalSteps = count($this->items);
```

Comment:
`$items` is pretty opaque here, maybe add a `getItems()` to the interface and describe what it should look like? I guess that's implementation specific, could be a class name, an array, etc. 

Do you have an example to illustrate what you could do with this?

Reply:
Sure, some generic examples:

* job which generates image variants (one item represents one image variant)
* job which migrates a list of objects (one item represents object ID)
* job which updates static cache (one item represents one URL)

[Specific example seen here.](https://github.com/silverstripe/silverstripe-assets/pull/397/files#diff-f84cb7facd19614bea691510775c0a49R9)

Reply:
Hmm I'm wondering how much value there is in defining an abstract class for this, with so much implicit variations. Maybe this could just be a "howto: implement multi step jobs" in the docs? Something that simply points out that all jobs have `$remaining` and `$totalSteps`.